src/server/scripts/Spells/spell_priest.cpp

```diff
@@ -88,6 +92,77 @@ class RaidCheck
         Unit const* _caster;
 };
 
+// -27811 - Blessed Recovery
+class spell_priest_blessed_recovery : public SpellScriptLoader
+{
+public:
+    spell_priest_blessed_recovery() : SpellScriptLoader("spell_priest_blessed_recovery") { }
+
+    class spell_priest_blessed_recovery_AuraScript : public AuraScript
+    {
+        PrepareAuraScript(spell_priest_blessed_recovery_AuraScript);
+
+        bool Validate(SpellInfo const* /*spellInfo*/) override
+        {
+            if (!sSpellMgr->GetSpellInfo(SPELL_PRIEST_BLESSED_RECOVERY_R1) 
+                || !sSpellMgr->GetSpellInfo(SPELL_PRIEST_BLESSED_RECOVERY_R2)
+                || !sSpellMgr->GetSpellInfo(SPELL_PRIEST_BLESSED_RECOVERY_R3))
+                return false;
+            return true;
+        }
+
+        bool CheckProc(ProcEventInfo& eventInfo)
+        {
+            if (eventInfo.GetSpellInfo())
+                if (eventInfo.GetSpellInfo()->SpellIconID == PRIEST_ICON_ID_BLESSED_RECOVERY)
```

Comment:
This entire check is not needed - script will only run for the intended talent because you are binding it to spell id in spell_script_names table